fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/ai/tasks/WorkerVillagerTask.java

```diff
@@ -0,0 +1,295 @@
+package net.fabricmc.fabric.api.object.builder.v1.ai.tasks;
+
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import net.minecraft.entity.ai.brain.BlockPosLookTarget;
+import net.minecraft.entity.ai.brain.MemoryModuleState;
+import net.minecraft.entity.ai.brain.MemoryModuleType;
+import net.minecraft.entity.ai.brain.WalkTarget;
+import net.minecraft.entity.ai.brain.task.FarmerVillagerTask;
+import net.minecraft.entity.ai.brain.task.Task;
+import net.minecraft.entity.passive.VillagerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.village.VillagerProfession;
+import net.minecraft.world.GameRules;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+//import org.jetbrains.annotations.Contract;
+//import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * A framework for creating simple world-affecting {@code VillagerTask}s, including pre-made functions for scanning for targets, etc.
+ * @apiNote Tasks may also extend {@code Task<VillagerEntity>}.
+ * @see FarmerVillagerTask FarmerVillagerTask
+ * @see net.minecraft.entity.ai.brain.task.BoneMealTask BoneMealTask
+ * @see Task
+ * @see VillagerTaskProviderRegistry
+ */
+public abstract class WorkerVillagerTask extends Task<VillagerEntity> {
```

Comment:
I don’t think this class is right for FAPI. It isn’t supposed to have helper but unneeded classes.

Reply:
No worries; I can drop it without any hassle.  I like the idea of helper classes, but I understand the necessity of a lightweight API.